mono/metadata/cominterop.c

```diff
@@ -1850,22 +1821,17 @@ mono_cominterop_release_all_rcws (void)
 }
 
 gpointer
-ves_icall_System_ComObject_GetInterfaceInternal (MonoComObject* obj, MonoReflectionType* type, MonoBoolean throw_exception)
+ves_icall_System_ComObject_GetInterfaceInternal (MonoComObjectHandle obj, MonoReflectionTypeHandle ref_type, MonoBoolean throw_exception, MonoError *error)
 {
 #ifndef DISABLE_COM
-	ERROR_DECL (error);
-	MonoClass *klass = mono_type_get_class (type->type);
-	if (!mono_class_init (klass)) {
-		mono_error_set_for_class_failure (error, klass);
-		mono_error_set_pending_exception (error);
+	MonoType * const type = MONO_HANDLE_GETVAL (ref_type, type);
+	MonoClass * const klass = mono_class_from_mono_type (type);
+	if (!mono_class_init_checked (klass, error))
 		return NULL;
-	}
 
-	gpointer itf = cominterop_get_interface_checked (obj, klass, error);
-	if (throw_exception)
-		mono_error_set_pending_exception (error);
-	else
-		mono_error_cleanup (error);
+	ERROR_DECL (error_ignored);
```

Comment:
Instead of creating a new error, can't you more simply conditionally cleanup the existing `error`? Something like the following would be equivalent to what you did without creating a temporary error.

```
gpointer const itf = cominterop_get_interface_checked (MONO_HANDLE_RAW (obj), klass, error);
if (!throw_exception)
	mono_error_cleanup (error);
```

Reply:
Maybe but not clearly/completely. I think you'd have to reinit. Which is a rare pattern, perhaps to be avoided?

It seems slightly better to perturb ignored extra state on the side, if intending to ignore, instead of mutating the "real" state temporarily, only to restore it.
Basically to reduce visible-or-even-temporary-maybe-not-visible side effects in general, if perf is ok.
It isn't clear cut though.

There is also the off chance of:
foo(error);
bar(ignore, error); // harmless to proceed a little?
if (ok(error)) // error from foo, unperturbed by bar.

Though given the scoped nature of icalls, this might be impossible to construct.

Reply:
As to  what is simpler is also subjective. What I have isn't complicated (subjectively) and is simpler to reason about, due to the reduced side effects.
What you suggest does save stack and perhaps perf, indeed.